client/clientservice/src/event_service.cpp

```diff
@@ -101,64 +107,69 @@ Status EventServiceImpl::Subscribe(ServerContext* context,
     if (not update) {
       continue;
     }
-    steady_clock::time_point start_processing = steady_clock::now(), end_processing;
-
-    if (std::holds_alternative<cc::EventGroup>(*update)) {
-      auto& event_group_in = std::get<cc::EventGroup>(*update);
-      EventGroup proto_event_group;
-      proto_event_group.set_id(event_group_in.id);
-      for (auto& event : event_group_in.events) {
-        proto_event_group.add_events(std::move(event));
-      }
-      *proto_event_group.mutable_record_time() = event_group_in.record_time;
-      *proto_event_group.mutable_trace_context() = {event_group_in.trace_context.begin(),
-                                                    event_group_in.trace_context.end()};
-
-      *response.mutable_event_group() = proto_event_group;
-      steady_clock::time_point start_write = steady_clock::now();
-      stream->Write(response);
-      metrics_.total_num_writes++;
-      // update write duration metric
-      end_processing = steady_clock::now();
-      auto duration_write = duration_cast<chrono::microseconds>(end_processing - start_write);
-      metrics_.write_duration.Get().Set(duration_write.count());
-    } else if (std::holds_alternative<cc::Update>(*update)) {
-      auto& legacy_event_in = std::get<cc::Update>(*update);
-      Events proto_events;
-      proto_events.set_block_id(legacy_event_in.block_id);
-      for (auto& [key, value] : legacy_event_in.kv_pairs) {
-        Event proto_event;
-        *proto_event.mutable_event_key() = std::move(key);
-        *proto_event.mutable_event_value() = std::move(value);
-        *proto_events.add_events() = proto_event;
+
+    {  // scoped timer 1 start
+      concord::diagnostics::TimeRecorder processing_duration_scoped_timer(*histograms_.processing_duration);
+      DurationTracker<chrono::microseconds> processing_dt("processing_dt", true);
+      if (std::holds_alternative<cc::EventGroup>(*update)) {
+        auto& event_group_in = std::get<cc::EventGroup>(*update);
+        EventGroup proto_event_group;
+        proto_event_group.set_id(event_group_in.id);
+        for (auto& event : event_group_in.events) {
+          proto_event_group.add_events(std::move(event));
+        }
+        *proto_event_group.mutable_record_time() = event_group_in.record_time;
+        *proto_event_group.mutable_trace_context() = {event_group_in.trace_context.begin(),
+                                                      event_group_in.trace_context.end()};
+        *response.mutable_event_group() = proto_event_group;
+        DurationTracker<chrono::microseconds> write_dt("write_dt", true);
+        stream->Write(response);
+        metrics_.write_duration.Get().Set(write_dt.totalDuration());
```

Comment:
On similar lines to legacy event update, do we need LOG_DEBUG statement here?

Reply:
I'll update here and the other place with LOG_DEBUG + the duration.
Thanks.